### PR TITLE
Allow fitting with a single data point and provide better error

### LINKED ISSUE #82: One element case not handled correctly
```julia
julia> xs = 10 .* rand(1)
1-element Vector{Float64}:
 7.534245260532203

julia> ys = sin.(xs) .+ 0.5 * rand(1)
1-element Vector{Float64}:
 1.32373419087167

julia> model = loess(xs, ys, span=0.5)
ERROR: MethodError: no method matching Loess.KDTree(::Matrix{Float64}, ::Vector{Int64}, ::Nothing, ::Set{Vector{Float64}}, ::Matrix{Float64})

Closest candidates are:
  Loess.KDTree(::Matrix{T}, ::Vector{Int64}, ::Loess.KDNode{T}, ::Set{Vector{T}}, ::Matrix{T}) where T<:AbstractFloat
   @ Loess ~/.julia/packages/Loess/YyInn/src/kd.jl:12
  Loess.KDTree(::AbstractMatrix{T}, ::Real, ::Real) where T<:AbstractFloat
   @ Loess ~/.julia/packages/Loess/YyInn/src/kd.jl:37

Stacktrace:
 [1] Loess.KDTree(xs::Matrix{Float64}, leaf_size_factor::Float64, leaf_diameter_factor::Int64)
   @ Loess ~/.julia/packages/Loess/YyInn/src/kd.jl:68
 [2] loess(xs::Matrix{Float64}, ys::Vector{Float64}; normalize::Bool, span::Float64, degree::Int64, cell::Float64)
   @ Loess ~/.julia/packages/Loess/YyInn/src/Loess.jl:70
 [3] loess
   @ ~/.julia/packages/Loess/YyInn/src/Loess.jl:43 [inlined]
 [4] #loess#10
   @ ~/.julia/packages/Loess/YyInn/src/Loess.jl:146 [inlined]
 [5] top-level scope
   @ REPL[26]:1

```

The loess estimate should just be the number itself.

### PR DESCRIPTION
**Pull request by andreasnoack**
message when data arrays are empty

Fixes #82

### CODE REVIEW

File: src/kd.jl [hunk 1]
```diff
@@ -9,11 +9,11 @@ end
 
 
 struct KDTree{T <: AbstractFloat}
-    xs::Matrix{T}         # A matrix of n, m-dimensional observations
-    perm::Vector{Int}     # permutation of data to avoid modifying xs
-    root::KDNode{T}       # root node
+    xs::Matrix{T}                   # A matrix of n, m-dimensional observations
+    perm::Vector{Int}               # permutation of data to avoid modifying xs
+    root::Union{Nothing, KDNode{T}} # root node
```
Comment (devmotion) [inline src/kd.jl:14]: Just to doublecheck, is this change needed to fix the bug?

Comment (andreasnoack) [reply src/kd.jl:14]: Yes. In the case of a single observations, the root with be a `nothing`.
